src/go/k8s/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -0,0 +1,234 @@
+package redpanda
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	cmapiv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/go-logr/logr/testr"
+	"github.com/redpanda-data/helm-charts/charts/redpanda"
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	"github.com/redpanda-data/helm-charts/pkg/helm/helmtest"
+	"github.com/redpanda-data/helm-charts/pkg/testutil"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/src/go/k8s/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/src/go/k8s/internal/testutils"
+	"github.com/redpanda-data/redpanda-operator/src/go/k8s/pkg/k3d"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+func TestRedpandaReconciler(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger := testr.New(t).V(0)
+	log.SetLogger(logger)
+	ctx = log.IntoContext(ctx, logger)
+
+	var cluster *k3d.Cluster
+	var err error
+	if val, ok := os.LookupEnv("EXISTING_K3D_CLUSTER"); ok && val != "" {
```

Comment:
Given the `-retain` flag and consistent naming what's the use case for being able to specify an existing cluster? Should `k3d.NewCluster` be updated to either delete or adopt any existing cluster?

Reply:
It was my quick hack to shorten developer cycle as I was iterating with reconciler logic. I'm all for settle on some convention. Would you like flag or env var that could pick existing k3d cluster?